### PR TITLE
Add experimental HIP support

### PR DESCRIPTION
**Pull request by G-071**
This PR adds the first experimental support for running Octo-Tiger on AMD GPUs. 

While the effort regarding a (performance-portable) AMD GPU port is still ongoing, I think this is a good point to get some of the changes into master (before the PR gets too unwieldy). 

Changelog:
- Added HIP kernel variants for most <sup>[1](#p2m)</sup> major CUDA kernels (basically by hipifying the API)
- Reworked CUDA/HIP Kernels for increased performance (separated Stencils into more blocks)
- Added support for running most  <sup>[1](#p2m)</sup> Kokkos kernels on the HIP backend via the hpx-kokkos hip executors
- Added CLI options to actually use the new kernels (device kernel types: HIP, KOKKOS_HIP)
- Added tests for all new kernels
- Added Jenkins Pipeline for AMD GPU tests on the Rostam MI100 nodes
- Added Jenkins Pipeline for the Stuttgart Radeon VII Pro node 
- Added warnings about HIP support still being experimental...

Note:
- Still does not quite achieve the same performance as we get on NVIDIA devices -> needs more work!
- CLI parameter names need to be renamed ("cuda_streams_per_gpu" is an odd name for the number of Kokkos / HIP executors..) - as this will change the interface for all existing scripts, I would postpone this task for a later PR

Build instructions:
- Set the cmake variable OCTOTIGER_WITH_CUDA=OFF and OCTOTIGER_WITH_HIP=ON
- Experimental toolchain branch available [here](https://github.com/STEllAR-GROUP/OctoTigerBuildChain/tree/jenkins-kamand)

Remaining ToDos
- [x] Cleanup

\
<a name="p2m">[1]</a>: Except the p2m kernel as yet, as it requires more effort to port but only constitutes a small percentage of the runtime

### CODE REVIEW

File: octotiger/profiler.hpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ struct timings
             std::cout << "   Regrid: "            << times_[time_regrid]  << " (" <<  100*times_[time_regrid] * tinv << " %)" << std::endl;
             std::cout << "   Computation + Regrid: "       << tcr << " (" <<  100*tcr * tinv << " %)" << std::endl;
         } else {
-            std::cout << "   Warning! Total time is 0! " << std::endl;
+            //std::cout << "   Warning! Total time is 0! " << std::endl;
```
Comment (diehlpk) [inline octotiger/profiler.hpp:94]: It seems we have some empty else block or?

Comment (G-071) [reply octotiger/profiler.hpp:94]: Actually yeah! We can comment in cout again.

Comment (diehlpk) [reply octotiger/profiler.hpp]: Sounds good.

File: test_problems/sphere/sphere.ini [hunk 2]
```diff
@@ -2,3 +2,4 @@ problem=solid_sphere
 max_level=3
 hydro=off
 disable_diagnostics=on
+stop_step=1
```
[the remark this answers is not in the document]
Comment (diehlpk) [reply test_problems/sphere/sphere.ini:5]: Sounds good.

**Review by diehlpk — APPROVED** [review]
LGTM!

**Pull request merged**